docs/source/core_specification/schema_types.rst

```diff
@@ -1,3 +1,282 @@
+
+Scalars
+-------
+
+The compiler uses the built-in GraphQL
+`scalar types <https://graphql.org/learn/schema/#scalar-types>`__ as well as three custom scalars:
+
+- :code:`DateTime` represents timezone-aware second-accuracy timestamps.
+- :code:`Date` represents day-accuracy date objects.
+- :code:`Decimal` is an arbitrary-precision decimal number object useful for representing values
+  that should never be rounded, such as currency amounts.
+
+Query Operation
+---------------
+
+GraphQL allows for three operation types *query*, *mutation* and *subscription*. The compiler
+only allows query operation types as shown in the code snippet below:
+
+.. code::
+
+    schema {
+        query: RootSchemaQuery
+    }
+
+A query may begin in any of the **root vertex types** specified by the :code:`RootSchemaQuery`
+object:
+
+.. code::
+
+    type RootSchemaQuery {
+        Animal: [Animal]
+        Entity: [Entity]
+        Food: [Food]
+        Species: [Species]
+        Toy: [Toy]
+    }s
+
+Inheritance
+-----------
+
+The compiler uses GraphQL interfaces and GraphQL unions in representing the inheritance structure
+of the underlying schema. You may ignore this section if are compiling to a language with no
+inheritance structure, (e.g. SQL).
+
+Interfaces
+~~~~~~~~~~
+
+GraphQL interfaces represent the abstract vertices of the underlying database.
+
+.. code::
+
+    interface Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+    }
+
+Abstract inheritance is modeled through interface implementation.
+
+.. code::
+
+    type Food implements Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+    }
+
+Unions and :code:`type_equivalence_hints`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+GraphQL does not support a notion of concrete inheritance. In other words, GraphQL objects cannot
+inherit from other objects. However, for certain query dialects, the compiler needs concrete
+inheritance information to emit the right query.
+
+To model concrete inheritance. we use GraphQL unions that encompass an object's
+subclasses and a :code:`type_equivalence_hints` parameter to signify that object is equivalent to
+the GraphQL union.
+
+For example, suppose :code:`Food` and :code:`Species` are concrete types and :code:`Food` is a
+superclass of :code:`Species` in an OrientDB schema. Then the GraphQL schema generation function
+would generate an union class in the schema
+
+.. code::
+
+    union Union__Food__Species = Food | Species
+
+and an entry in :code:`type_equivalence_hints` mapping :code:`Food` to
+:code:`Union_Food_Species`.
+
+
+Type coercions
+~~~~~~~~~~~~~~
+
+Type coercions are operations than can be run against interfaces and unions to create a new scope
+whose type is different than the type of the enclosing scope of the coercion. Type coercions are
```

Comment:
This first sentence is pretty thick, maybe add a TODO to clarify it.

Reply:
Will do.